Content.Client/Wires/Visualizers/WiresVisualizerSystem.cs

```diff
@@ -0,0 +1,33 @@
+using Content.Shared.Wires;
+using Robust.Client.GameObjects;
+
+namespace Content.Client.Wires.Visualizers
+{
+    public sealed class WiresVisualizerSystem : VisualizerSystem<WiresVisualsComponent>
+    {
+        [Dependency] private readonly AppearanceSystem _appearanceSystem = default!;
+
+        protected override void OnAppearanceChange(EntityUid uid, WiresVisualsComponent component, ref AppearanceChangeEvent args)
+        {
+            if (args.Sprite == null)
+                return;
+
+            var layer = args.Sprite.LayerMapReserveBlank(WiresVisualLayers.MaintenancePanel);
+
+            if (_appearanceSystem.TryGetData(uid, WiresVisuals.MaintenancePanelState, out var panelState, args.Component))
```

Comment:
you can also get the data from AppearanceChangeEvent, in the AppearanceData field

Reply:
Hah, good to know, all this fussing about with visualizer and appearance data and I never bothered to have a proper look at the event itself. Getting that data from the event now, removed the unnecessary dependency and while I was at I also made the typecast of the appearance data a bit safer than just optimistically relying on it being the type we expect.

Also going to make similar changes to the vending machine PR where I also used the AppearanceSystem and ended up making the typecast needlessly complicated in retrospect.

Reply:
sounds good to me 👍